tests/sentry/event_manager/test_event_manager.py

```diff
@@ -633,6 +635,119 @@ def test_multiple_pending_commit_resolution(self):
+
+        # commit that resolved the issue is part of a PR, but all commits within the PR are unreleased
+
+        commit = Commit.objects.create(
+            organization_id=group.project.organization_id, repository_id=repo.id, key="a" * 40
+        )
+
+        second_commit = Commit.objects.create(
+            organization_id=group.project.organization_id, repository_id=repo.id, key="b" * 40
+        )
+
+        GroupLink.objects.create(
+            group_id=group.id,
+            project_id=group.project_id,
+            linked_type=GroupLink.LinkedType.commit,
+            linked_id=commit.id,
+            relationship=GroupLink.Relationship.resolves,
+        )
+
+        pr = PullRequest.objects.create(
+            organization_id=group.project.organization_id,
+            repository_id=repo.id,
+            key="1",
+        )
+
+        PullRequestCommit.objects.create(pull_request_id=pr.id, commit_id=commit.id)
+        PullRequestCommit.objects.create(pull_request_id=pr.id, commit_id=second_commit.id)
+
+        assert PullRequestCommit.objects.filter(pull_request_id=pr.id, commit_id=commit.id).exists()
+        assert PullRequestCommit.objects.filter(
+            pull_request_id=pr.id, commit_id=second_commit.id
+        ).exists()
+
+        assert not ReleaseCommit.objects.filter(commit__pullrequestcommit__id=commit.id).exists()
+        assert not ReleaseCommit.objects.filter(
+            commit__pullrequestcommit__id=second_commit.id
+        ).exists()
+
+        pending = has_pending_commit_resolution(group)
+        assert pending
+
+    def test_has_pending_commit_resolution_issue_regression_released_commits(self):
+        project_id = 1
+        event = self.make_release_event("1.0", project_id)
+        group = event.group
+        release = self.create_release(project=self.project, version="1.1")
+
+        repo = self.create_repo(project=group.project)
+
+        # commit 1 is part of the PR, it resolves the issue in the commit message, and is unreleased
+        commit = Commit.objects.create(
+            organization_id=group.project.organization_id, repository_id=repo.id, key="a" * 38
+        )
+
+        GroupLink.objects.create(
+            group_id=group.id,
+            project_id=group.project_id,
+            linked_type=GroupLink.LinkedType.commit,
+            linked_id=commit.id,
+            relationship=GroupLink.Relationship.resolves,
+        )
+
+        # commit 2 is part of the PR, but does not resolve the issue, and is released
+        released_commit = Commit.objects.create(
+            organization_id=group.project.organization_id, repository_id=repo.id, key="b" * 38
+        )
+
+        # commit 3 is part of the PR, but does not resolve the issue, and is unreleased
+        unreleased_commit = Commit.objects.create(
+            organization_id=group.project.organization_id, repository_id=repo.id, key="c" * 38
+        )
+
+        assert Commit.objects.all().count() == 3
```

Comment:
It might make sense to put all your asserts at the end of the test.

Reply:
I put the assert after the relevant line that I was testing since there are a lot of asserts - I thought it might make it clearer. Is it conventional to put them all at the end?

Reply:
I think this might be debatable. I generally try to follow this pattern:
https://automationpanda.com/2020/07/07/arrange-act-assert-a-pattern-for-writing-good-tests/

I'm sure others might have differing opinions on this.